chainio/clients/elcontracts/reader.go

```diff
@@ -515,6 +515,92 @@ func (r *ChainReader) GetClaimerFor(
 	return r.rewardsCoordinator.ClaimerFor(&bind.CallOpts{Context: ctx}, earner)
 }
 
+// Returns the submission nonce for an avs
+func (r *ChainReader) GetSubmissionNonce(
+	ctx context.Context,
+	avs gethcommon.Address,
+) (*big.Int, error) {
+	if r.rewardsCoordinator == nil {
+		return nil, errors.New("RewardsCoordinator contract not provided")
+	}
+	return r.rewardsCoordinator.SubmissionNonce(&bind.CallOpts{Context: ctx}, avs)
+}
+
+// Returns whether a hash is a valid rewards submission hash for a given avs
+func (r *ChainReader) GetIsAVSRewardsSubmissionHash(
+	ctx context.Context,
+	avs gethcommon.Address,
+	hash [32]byte,
+) (bool, error) {
+	if r.rewardsCoordinator == nil {
+		return false, errors.New("RewardsCoordinator contract not provided")
+	}
+	return r.rewardsCoordinator.IsAVSRewardsSubmissionHash(&bind.CallOpts{Context: ctx}, avs, hash)
+}
+
+// Returns whether a hash is a valid rewards submission for all hash for a given avs
+func (r *ChainReader) GetIsRewardsSubmissionForAllHash(
+	ctx context.Context,
+	avs gethcommon.Address,
+	hash [32]byte,
+) (bool, error) {
+	if r.rewardsCoordinator == nil {
+		return false, errors.New("RewardsCoordinator contract not provided")
+	}
+	return r.rewardsCoordinator.IsRewardsSubmissionForAllHash(&bind.CallOpts{Context: ctx}, avs, hash)
+}
+
+// Returns whether a submitter is a valid rewards for all submitter
+func (r *ChainReader) GetIsRewardsForAllSubmitter(
+	ctx context.Context,
+	submitter gethcommon.Address,
+) (bool, error) {
+	if r.rewardsCoordinator == nil {
+		return false, errors.New("RewardsCoordinator contract not provided")
+	}
+	return r.rewardsCoordinator.IsRewardsForAllSubmitter(&bind.CallOpts{Context: ctx}, submitter)
+}
+
+// Returns whether a hash is a valid rewards submission for all earners hash for a given avs
+func (r *ChainReader) GetIsRewardsSubmissionForAllEarnersHash(
+	ctx context.Context,
+	avs gethcommon.Address,
+	hash [32]byte,
+) (bool, error) {
+	if r.rewardsCoordinator == nil {
+		return false, errors.New("RewardsCoordinator contract not provided")
+	}
+	return r.rewardsCoordinator.IsRewardsSubmissionForAllEarnersHash(&bind.CallOpts{Context: ctx}, avs, hash)
+}
+
+// Returns whether a hash is a valid operator set performance rewards submission hash for a given avs
```

Comment:
This doc is the same as `GetIsOperatorDirectedOperatorSetRewardsSubmissionHash`. What's the difference here? I checked the contracts and the documentation is the same for both fields, so let's keep it as-is. I leave this comment here for posterity 😞

Reply:
Haha thats what the comments are in the smart contract:
https://github.com/Layr-Labs/eigenlayer-contracts/blob/499fd45503de7a5fe24b76d7fe1231dea18ac12d/src/contracts/core/RewardsCoordinatorStorage.sol#L120

https://github.com/Layr-Labs/eigenlayer-contracts/blob/499fd45503de7a5fe24b76d7fe1231dea18ac12d/src/contracts/core/RewardsCoordinatorStorage.sol#L132

Reply:
Yeah I just copied it over from there